File: kattis/geppetto.cpp
// vicfred
// https://open.kattis.com/problems/geppetto
// brute force, bitmask
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N, M;
    cin >> N >> M;

    vector<int> a(M), b(M);
    for(int i = 0; i < M; i++) {
        cin >> a[i] >> b[i];
        a[i] -= 1;
        b[i] -= 1;
    }

    long long ans = 0LL;
    for(int mask = 0; mask < 1 << N; mask++) {
        bool flag = true;

        for(int i = 0; i < M; i++) {
            if(
                mask & (1 << a[i]) and
                mask & (1 << b[i])
            ) {
                flag = false;
                break;
            }
        }
        
        if(flag)
            ans += 1;
    }

    cout << ans << endl;

    return 0;
}

File: kattis/geppetto.py
# Vicfred
# https://open.kattis.com/problems/geppetto
# brute force, bitmask
N, M = map(int, input().split())

a = [0]*M
b = [0]*M

for i in range(M):
    x, y = map(int, input().split())
    a[i] = x-1
    b[i] = y-1

ans = 0
for mask in range(1<<N):
    flag = True
    
    for i in range(M):
        if mask & (1 << a[i]) and mask & (1 << b[i]):
            flag = False
            break

    if flag:
        ans += 1

print(ans)

